model_v4.py

```diff
@@ -0,0 +1,382 @@
+        setattr(self, f'inception_{addition_name}_1x1_act',
+                Activation('relu', name=f'inception_{addition_name}_1x1_act')(
+                    getattr(self, f'inception_{addition_name}_1x1_bn')))
+
+    def create_interception_block(self, input_layer, addition_name='',
+                                    conv1_5x5=None, conv2_5x5=None,
+                                    conv1_3x3=None, conv2_3x3=None,
+                                    conv_filters=None, pool_filters=None,
+                                    pool_type='max', addition_pool=False):
+        """
+        Create inception block
+        :param input_layer: the last layer
+        :param addition_name:  index in the naming system (idx of inception layer)
+        :param conv1_5x5: num of filters in conv1_5x5 layer; None if that layer doesn't need
+        :param conv2_5x5: num of filters in conv2_5x5 layer; None if that layer doesn't need
+        :param conv1_3x3: num of filters in conv1_3x3 layer; None if that layer doesn't need
+        :param conv2_3x3: num of filters in conv2_3x3 layer; None if that layer doesn't need
+        :param conv_filters: num of filters in conv2 layer; None if that layer doesn't need
+        :param pool_filters: num of filters in conv1 layer; None if that layer doesn't need
+        :param pool_type:  'max' - use MaxPooling or 'avg' - use AveragePooling
+        :param addition_pool: add a stream with MaxPooling layer
+        :return: None
+        """
+        layers_set_list = []
+        strides = (1, 1) if not addition_pool else (2, 2)
+
+        if conv_filters is not None and pool_filters is not None:
+            self.create_conv_pool_layers_set(input_layer, addition_name=addition_name, conv_filters=conv_filters,
+                                                          pool_filters=pool_filters, pool_type=pool_type)
+            layers_set_list.extend([getattr(self, f'inception_{addition_name}_1x1_act'),
+                                    getattr(self, f'inception_{addition_name}_pool_act')])
+
+        if conv1_5x5 is not None and conv2_5x5 is not None:
+            setattr(self, f'inception_{addition_name}_5x5_conv1',
+                          Conv2D(filters=conv1_5x5,
+                                 kernel_size=(1, 1),
+                                 padding='same',
+                                 kernel_regularizer=l2(0.01),
+                                 name=f'inception_{addition_name}_5x5_conv1')(input_layer))
+            setattr(self, f'inception_{addition_name}_5x5_bn1',
+                    BatchNormalization(name=f'inception_{addition_name}_5x5_bn1')(
+                        getattr(self, f'inception_{addition_name}_5x5_conv1')))
+            setattr(self, f'inception_{addition_name}_5x5_act1',
+                    Activation('relu', name=f'inception_{addition_name}_5x5_act1')(
+                        getattr(self, f'inception_{addition_name}_5x5_bn1')))
```

Comment:
_setattr_ and _getattr_ should be used only for introspection tasks. You don't need it here.